### PR TITLE
Fixes #193 - Adds better formatting for the RedBoxDialog.

### LINKED ISSUE #193: Improve XAML formatting for Red Box
Try to match iOS and Android as closely as possible.

### PR DESCRIPTION
**Pull request by rozele**
Replacing with a XAML view with list binding to make red box formatting quicker and easier (and more declarative).

### CODE REVIEW

File: ReactWindows/ReactNative/ReactNative.csproj [hunk 1]
```diff
@@ -173,8 +173,11 @@
     <Compile Include="DevSupport\ProgressDialog.xaml.cs">
       <DependentUpon>ProgressDialog.xaml</DependentUpon>
     </Compile>
-    <Compile Include="DevSupport\RedBoxDialog.cs" />
+    <Compile Include="DevSupport\RedBoxDialog.xaml.cs">
+      <DependentUpon>RedBoxDialog.xaml</DependentUpon>
+    </Compile>
     <Compile Include="DevSupport\ShakeAccelerometer.cs" />
+    <Compile Include="DevSupport\StackFrameConverter.cs" />
     <Compile Include="DevSupport\StackFrameExtensions.cs" />
```
Comment (rozele) [inline ReactWindows/ReactNative/ReactNative.csproj]: Delete?

File: ReactWindows/ReactNative/DevSupport/DevSupportManager.cs [hunk 2]
```diff
@@ -262,17 +263,17 @@ public void UpdateJavaScriptError(string title, JArray details, int errorCookie)
                 }
 
                 _redBoxDialog.Title = title;
-                _redBoxDialog.StackTrace = StackTraceHelper.ConvertJavaScriptStackTrace(details);
+                _redBoxDialog.StackTrace = StackTraceHelper.ConvertJavaScriptStackTrace(details).ToList();
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevSupportManager.cs]: Remove this?

File: ReactWindows/ReactNative/DevSupport/StackTraceHelper.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@ public static IStackFrame[] ConvertJavaScriptStackTrace(JArray stack)
 
         public static IStackFrame[] ConvertNativeStackTrace(Exception exception)
         {
-            var stackTrace = new StackTrace(exception, true);
+            var stackTrace = new StackTrace(exception, false);
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/StackTraceHelper.cs]: revert to true.

File: ReactWindows/ReactNative/DevSupport/StackFrameExtensions.cs [hunk 4]
```diff
@@ -10,19 +10,16 @@ public static string PrettyPrint(this IStackFrame[] stackTrace)
...
             {
                 stringBuilder
-                    .AppendLine(frame.Method)
+                    .AppendLine(frame.Method);
+
+                var fileName = frame.FileName ?? "<filename unknown>";
+                stringBuilder
                     .Append("    ")
-                    .Append(frame.FileName)
+                    .Append(fileName)
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/StackFrameExtensions.cs]: Replace with SourceInfo

**Comment by mpodwysocki** [comment]
LGTM :+1:

**Pull request merged**